### ISSUE TITLE
How to determine F3, F4 RS-485 polarity (A,B)

### ISSUE THREAD

**Issue by guybridge** [DannyDeGaspari/Samsung-HVAC-buscontrol#13]
Hi,

I'm trying to figure out which way the RS-485 module's A, B poles should be hooked up to F3, and F4 from the unit.

So far i've tried both ways but don't see any traffic from the serial_dump.py

Just trying to rule out the basics.

**Comment by windymonkey** [comment]
F1F2 is RS485 communication. while F3F4 is 12VDC power line communication

**Comment by guybridge** [comment]
On mine it doesn't have F1,F2. Only V1,V2 which is 12v.

F3,4 appears to be the comms.

![IMG_20220613_091118_787.jpg](https://user-images.githubusercontent.com/10903500/173262726-01bd8ee8-876e-43e9-bd9e-6df59387b384.jpg)![IMG_20220613_091110_169.jpg](https://user-images.githubusercontent.com/10903500/173262732-bbb8e84d-db9b-480c-ba73-b3cc9e095966.jpg)

**Comment by windymonkey** [comment]
Yup, V1V2 is a 12 VDC power supply. I mean F1F2 is RS485 base comm on indoor/outdoor unit. Please use an oscilloscope with F3F4 terminal and see the magic :D
![z3487961300326_b618b817080a13659adb29af4a94798f](https://user-images.githubusercontent.com/7168911/173263102-d6c0bc85-9b8c-48ce-b19d-5f7e6ce3e620.jpg)

**Comment by guybridge** [comment]
Ok cool, yep don't have one handy. So If I had one I could determine the polarity for A and B?

**Comment by windymonkey** [comment]
Nope, F3F4 is power line communication not RS485. It means both power supply and communication share the same F3F4 terminals. You can try unplugging V1V2 and see if your remote is still working. I made a gateway to act as slave remote while WE10N is acting as master remote. Quite sure about this.

**Comment by guybridge** [comment]
I see.

Unplugged the V1,2 and power but it causes the display to turn off.

**Comment by guybridge** [comment]
I don't mind removing the wired remote completely. Really I wanted replace completely with the Pi/RS-485 unit.

When I plug the RS-485 with in PI into F3, and F4 - I can't see any data but if I send a command like power on

It causes the current remote control to flash red/green and say E601 which I believe is comms error so I thought I was on the right track with the new module.

**Comment by windymonkey** [comment]
Let me explain more, maybe my remote version is different from yours, and I don't know your wiring.
In my case, the remote connects directly to the indoor unit via F3F4 (12VDC), and no need V1V2 terminal for power. And the F3F4 is not RS485 interface. 
They using PLC 
I'm from Vietnam, maybe yours remote using RS485, please unplug the white plastic cover, I can see what kind of IC they using on your remote PCB.
And E601 error means no communication between your remote and the indoor unit.
Please see the image above.

**Comment by DannyDeGaspari** [comment]
On my system (Belgium) F1 to F4 have following meaning: https://github.com/DannyDeGaspari/Samsung-HVAC-buscontrol#physical-layer
The F1 and F2 connections are only available on the indoor units, not on the wired remote. It is the direct connection to the outdoor unit.
I also have the V1 and V2 connections on the wired remote and it is 12V power supply.

**Comment by a1studmuffin** [comment]
@windymonkey Thank you for this information - I think I have the same setup as you (MWR-WE10N) with only F3+F4 terminals (no V1+V2) that carry 12V power and data on the same line. My understanding was this is the NASA protocol that the newer Samsung models use. If you could provide any additional information on how you got your setup working with this protocol, it would be greatly appreciated! I'd like to get control of my Samsung AC via esphome + Home Assistant. I'm confident with the home automation side of things, but not so much with this NASA protocol.

**Comment by lanwin** [comment]
Did someone of you tried to attach directly to F1,F2 and make it work? Meaning putting the Pi directly behind the outdoor unit?

**Comment by lanwin** [comment]
> On mine it doesn't have F1,F2. Only V1,V2 which is 12v.
> 
> F3,4 appears to be the comms.
> 
> ![IMG_20220613_091118_787.jpg](https://user-images.githubusercontent.com/10903500/173262726-01bd8ee8-876e-43e9-bd9e-6df59387b384.jpg)![IMG_20220613_091110_169.jpg](https://user-images.githubusercontent.com/10903500/173262732-bbb8e84d-db9b-480c-ba73-b3cc9e095966.jpg)

Yep that are the new NASA based connection. I have them too. Do you get any data with it yet?

**Comment by lanwin** [comment]
I wonder if the NASA devices really still uses RS-485. Look at that device 

https://www.mtf-online.net/at/produkte/samsung/fjm-systeme/samsung-fjm-systeme-produkt?tx_mtfproductdb_mtfproductdisplay%5Baction%5D=show&tx_mtfproductdb_mtfproductdisplay%5Bcontroller%5D=Product&tx_mtfproductdb_mtfproductdisplay%5Bproduct%5D=1129&cHash=6ddf6f54579b9c46c908ea02f1bf9c20

Why is such a "complicated" device needed to convert between the old RS-485 and NASA.

**Comment by amitra93** [comment]
> @windymonkey Thank you for this information - I think I have the same setup as you (MWR-WE10N) with only F3+F4 terminals (no V1+V2) that carry 12V power and data on the same line. My understanding was this is the NASA protocol that the newer Samsung models use. If you could provide any additional information on how you got your setup working with this protocol, it would be greatly appreciated! I'd like to get control of my Samsung AC via esphome + Home Assistant. I'm confident with the home automation side of things, but not so much with this NASA protocol.

I have a similar setup (apartment with 3 rooms, 1 MWR-WE10N in each room, only F3 and F4 available, F1/F2 and V1/V2 are not available. Did you ever get this to work with Home Assistant?

**Comment by a1studmuffin** [comment]
> I have a similar setup (apartment with 3 rooms, 1 MWR-WE10N in each room, only F3 and F4 available, F1/F2 and V1/V2 are not available. Did you ever get this to work with Home Assistant?

I did! However I gave up on the idea of trying to reverse engineer the NASA protocol - that's a little beyond my current skillset. Instead I bought a MIM-H04AN wifi kit ($300, urgh) and integrated that with Home Assistant.

If you go down this path you can just use the official SmartThings integration, however you'll need SSL or a Home Assistant cloud subscription for SmartThings to communicate with your Home Assistant instance.

I wasn't keen on setting all that up just for this one integration, so I rolled my own custom component that uses smartthings-cli under the hood: https://github.com/a1studmuffin/samsungac-cli

It still requires the cloud unfortunately, but at least saved me the pain of SSL certificates/reverse proxy etc. One day I'll get it off the cloud... one day... but at least my wife is happy in the interim. :)

**Comment by lanwin** [comment]
Well I have reverse engineered the NASA protocol and it works on an ESP with ESPHome and is directly integrated into Home Assist. https://github.com/lanwin/esphome_samsung_ac

I also adding NonNASA support to that solution and it works directly on the device communication bus.

**Comment by a1studmuffin** [comment]
Wow, that's an amazing effort @lanwin! That must have taken some time to figure out. I'll definitely try migrating to this at some point.
